src/geometry/manifold/algebra/smooth_functions.lean

```diff
@@ -238,6 +238,12 @@ instance algebra : algebra 𝕜 C^∞⟮I, N; 𝓘(𝕜, A), A⟯ :=
   smul_def' := λ c f, by ext x; exact algebra.smul_def' _ _,
   ..smooth_map.semiring }
 
+/-- A special case of `pi.algebra` for non-dependent types. Lean get stuck on the definition
+below without this. -/
+instance _root_.function.algebra (I : Type*) {R : Type*} (A : Type*) {r : comm_semiring R}
+  [semiring A] [algebra R A] : algebra R (I → A) :=
+pi.algebra _ _
```

Comment:
I introduced `function.has_scalar` and `function.smul_comm_class` in another recent PR, and discussed with @madvorak about adding `function.has_add` etc to fix a similar problem - I'd prefer to leave collecting all those instances together into the right place to a follow-up PR.